Add endpoint aggregation capability

nsavoire opened the pull request:
* new function `setEndPoint` in timeprofiler allows to set endpoint for a given root span id.
* Timeprofiler records the mapping and use it during serialization to add `trace endpoint` labels for samples whose root span id is in the mapping.

szegedi reviewed (approved):
This looks good to me, although I'm curious about the further implementation in the tracer. Do we know when can we call `setEndPoint` in the tracer?

szegedi commented on ts/src/profile-serializer.ts:
```diff
@@ -344,7 +345,23 @@ export function serializeTimeProfile(
+  labelSet: LabelSet,
+  rootSpanIdToEndPointMap: Map<string, string>
+) {
+  const localRootSpanId = labelSet['local root span id'];
```
On second thought, it does make me sad that we need to veer away from having a generic solution for "labels" and push down knowledge about our specific single usage of this feature into the profiler codebase.

nsavoire replied:
Changed the API to be more generic !

szegedi commented on bindings/profilers/wall.cc:
```diff
@@ -243,18 +241,16 @@ class SignalHandler {
   }
 
   auto isolate = Isolate::GetCurrent();
-  // auto labelKey = Nan::New<String>("label").ToLocalChecked();
```
Ouch, I might've left this one in… thanks for catching it

nsavoire replied:
That's might have been me when debugging some issue ;)

szegedi replied:
Funny, I swear I had the same debugging line some weeks ago 😂

szegedi reviewed (approved):
All good, two minor points though:
* I would've appreciated a separate commit with just the nomenclature change (labels -> context), that would've made the significant changes stand out better.
* We can keep mutateContext, but if we don't think we'll use it, removing it is also an option.

nsavoire commented on bindings/profilers/wall.cc:
```diff
@@ -629,49 +660,86 @@ v8::CpuProfiler* WallProfiler::CreateV8CpuProfiler() {
+  curContext_.store(newCurContext, std::memory_order_relaxed);
+}
+
+bool WallProfiler::MutateContext(v8::Isolate* isolate,
```
I removed `MutateContext`.

nsavoire commented:
> All good, two minor points though:
> 
> * I would've appreciated a separate commit with just the nomenclature change (labels -> context), that would've made the significant changes stand out better.
> * We can keep mutateContext, but if we don't think we'll use it, removing it is also an option.

I split the PR into 3 commits and removed `mutateContext`.

szegedi commented:
> I split the PR into 3 commits and removed `mutateContext`.

Thank you, although I didn't expect it for this PR retroactively – I have already reviewed it in its original form after all :-) Just something to keep in mind as a good practice going forward when doing these kinds of lexical changes that touch many lines.
